src/tcgwars/logic/impl/gen8/BattleStyles.groovy

```diff
@@ -1995,7 +2001,9 @@ public enum BattleStyles implements LogicCardInfo {
                 damage 60
               }
             }
-            top.discard()
+            afterDamage {
+              top.discard()
```

Comment:
You might want to clarify this one. Seems impossible to discard after damage when discarding calculates the damage.